Resolved issues for document "IHE_Suppl_DEV_Profile_Monitored_Communication.docx"

PeterKranich opened the pull request:
Resolved issues for document "IHE_Suppl_DEV_Profile_Monitored_Communication.docx"; new version 1.4


<!-- If this pull request closes an issue, please mention the issue number below -->
Closes #28, #29, #30

## 📑 Description
* Removed transaction DEV54 and reformatted sections accordingly
* Updated message forrmat requirement: utilized HL7 message structure shall be ORU_R44
* Updated the message examples according to the new message structure
* Updated version to 1.4 and date to 15-July-2025

<!-- You can also choose to add a list of changes and if they have been completed or not by using the markdown to-do list syntax
- [ ] Not Completed
- [x] Completed
-->

## ✅ Checks
<!-- Make sure your pr passes the CI checks and do check the following fields as needed - -->
- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [x] All the tests have passed
- [x] I have selected a committee co-chair to review the PR

## ℹ Additional Information
All changes are marked with change bars in the document

PeterKranich committed:
Resolved issues #28, #29, #30 for document "IHE_Suppl_DEV_Profile_Monitored_Communication.docx"; new version 1.4

JavierEspina reviewed (commented):
1. In Table 14.1-1 I would expect to see a row for the reporter and a row for the consumer, both supporting DEV-53, the former as initiator and the latter as responder
2. In Figure 14.1-1 the "Acknowledge Heartbeat Message" line could be deleted. In this kind actor diagrams (for other DEV TF Profiles) acknowledgements are not typically depicted. They are in flow charts (such as Figure 14.4.2.2-1 or Figure 3.53.4-1, where the depicted acks definitely make sense).

Otherwise the changes made appear to correctly address DEV (PCMC) issues 28-30. Review approval by either  Kurt or Monroe or Rob recommended.

monroepattillo commented:
In section 14.4.2.1 Use Case #1: Reliable Alert Distribution, replace the sentence with something more aligned with the ACM profile, such as...
Reported alerts (physiologic alarms, technical alarms, advisories) from an Alert Communication Management (ACM) Alert Reporter (AR) actor needs to be disseminated reliably to ACM Alert Communicator (AC) associated endpoint communication devices (various types of pagers, phones, tablets, badges, watches, computers, etc.) as actionable, for message and procedural response by doctors, clinicians, and other alert event assigned staff.

monroepattillo commented:
Section 14.4.2.1.2 Reliable Alert Distribution Process Flow
Not sure the ACM actors should be referenced without associated with the PCMC actors.  Maybe something like the ACM AR actor acting as a PCMC PCD REPORTER actor sends the heartbeat message.  And the ACM AM actor acting as a PCMC PCD CONSUMER actor responds to the heartbeat message.

monroepattillo commented:
Section 14.5 PCMC Security Consideration
Replace paragraph with common IHE Devices ITI domain associated security considerations paragraph.

monroepattillo commented:
Section 3.53.4.1 Send Heartbeat Message [DEV-53]
DEV-53 is now its own transaction.  It is no longer an "in-line" alert (ACM PCD-04) event.

monroepattillo commented:
Section 3.53.4.1.2 Message Semantics, line 479
Another old reference to "in-line" alert.

monroepattillo commented:
Line 485, segment OBX Field content table, row OBX-11, Note column, add justification for "F" value, such as...
Final (F) is indicated so as to avoid encumbering clinical staff with review and confirmation effort for a system-to-system internal event that is not patient physiology associated and occurs in high volumes over the course of a staff shift.

monroepattillo commented:
FYI - While the alarm standard indicates alarm annunciation to a responsible clinician within 10 seconds, deployments for physiologic alarm notifications for life critical events (asystole) are typically 5 seconds or less, and the ACM AM typically only receives the PCD-04 Report Alert from the AR after about 5 of those 10 seconds so that the physiologic monitor can create the before to event to after ECG waveform, leaving the ACM solution (AM, AC, endpoint) about 2 seconds to get the alert notification to an actionable clinician.  If the alert dissemination to the primary recipient is undeliverable or is not endpoint acknowledged as received within 2 seconds or is not responded to by the clinician (Accept, Reject) within 2 seconds, the AM sends it to the secondary which accounts for 6 out of 10 seconds.

PeterKranich commented:
> 1. In Table 14.1-1 I would expect to see a row for the reporter and a row for the consumer, both supporting DEV-53, the former as initiator and the latter as responder
> 2. In Figure 14.1-1 the "Acknowledge Heartbeat Message" line could be deleted. In this kind actor diagrams (for other DEV TF Profiles) acknowledgements are not typically depicted. They are in flow charts (such as Figure 14.4.2.2-1 or Figure 3.53.4-1, where the depicted acks definitely make sense).
> 
> Otherwise the changes made appear to correctly address DEV (PCMC) issues 28-30. Review approval by either Kurt or Monroe or Rob recommended.

@JavierEspina I agree with #1. In the last WG meeting I was told to leave the acknowledge HB Msg action. Let's discuss this in the next review meeting.

PeterKranich commented:
> Section 14.5 PCMC Security Consideration Replace paragraph with common IHE Devices ITI domain associated security considerations paragraph.

@monroepattillo: I copied that from the PCIM profile:
7.5 PCIM Security Considerations
This profile itself does not impose specific requirements for authentication, encryption, or 430 auditing, leaving these matters to site-specific policy or agreement based on careful risk analysis taking into account the security and privacy sensitivity of the patient and device-patient association content being handled. The IHE PCD Technical Framework identifies security requirements across all PCD profiles.

PeterKranich commented:
> Section 3.53.4.1 Send Heartbeat Message [DEV-53] DEV-53 is now its own transaction. It is no longer an "in-line" alert (ACM PCD-04) event.

@monroepattillo: I was not sure about that: yes, it is its own transaction, but it is formatted as an "in-line" alert event!?

kelliaso commented:
Line 485: In the table on page 23, OBX-8 has 3 values for abnormal flags.  Use of OBX-8 components has been deprecated with new implementations using separate OBX segments.  As of HL7 2.7, OBX-8 has been changed to Interpretation Codes.

ToddCooper commented:
@kelliaso ...

> We need to review the Actor names, they should be what the actor is doing, not include Profile name. PCMC Actors could be Device Heartbeat Reporter, Device Heartbeat Consumer. Similar to Device Observation Reporter, Device Observation Consumer.

My guess is that the DCC review (think Kevin O'D.) would encourage even shortening it to Heartbeat Reporter / Consumer (or similar) since technically, actor names should be reusable across profiles, including in other domains!  Of course, this works well too.

PeterKranich commented:
> Line 485, segment OBX Field content table, row OBX-11, Note column, add justification for "F" value, such as... Final (F) is indicated so as to avoid encumbering clinical staff with review and confirmation effort for a system-to-system internal event that is not patient physiology associated and occurs in high volumes over the course of a staff shift.

@monroepattillo: This is also defined by the "in-line" event format definition, but it is fine for me to add your proposed note - I added it for the next version.

PeterKranich commented:
> We need to review the Actor names, they should be what the actor is doing, not include Profile name. PCMC Actors could be Device Heartbeat Reporter, Device Heartbeat Consumer. Similar to Device Observation Reporter, Device Observation Consumer.

@kelliaso, @ToddCooper: fine with me. The original intention was to insert the Heartbeat as an "in-line" event in other IHE profiles but my understanding now is that we all tend to a separate Heartbeat message. In this case, we can define more specific actor names.

PeterKranich committed:
Incorporated some of the comments from @JavierEspina and @monroepattillo.

PeterKranich committed:
Updated document after the IHE ACM WG meeting on 25-Sep-2025.

PeterKranich committed:
Changed transaction name, describtions, and acronyms upon Monroe's request

PeterKranich commented:
> While the PCMC profile primarily describes bidirectional communication between Heartbeat Reporter and Heartbeat Consumer, many medical devices, especially legacy, low-power, or highly secure systems, may only support unidirectional communication, where the device can send data but cannot receive commands or acknowledgments.
> 
> So, maybe we can consider “Start-Only Watchdog Model for Unidirectional Communication”?
> 
> In unidirectional environments, the “start_only” event phase of the heartbeat message can be used as a periodic watchdog signal. Each signal acts as a discrete confirmation of device health within a predefined timeout window. The Heartbeat Consumer monitors the arrival of these signals and infers device status based on their presence or absence, triggering alerts if expected signals are missed.
> 
> I understand the limitations of such an implementation. For example, the periodicity and timeout values should be carefully selected based on the criticality of the device and the clinical workflow requirements. However, I think this approach could broaden the applicability of PCMC across a wider range of medical device connectivity scenarios.

@GongliW: Thanks for your comments. IMHO, no mutual reliable communication can be established in this case. When a low-power sensor wakes up every 5 minutes and reports its data (vital signs, alerts, etc.) to the consumer, but cannot receive the state of the consumer, it does not know if the consumer can process the data as intended (e.g. disseminate alerts to the caregiver's smart phones) between the reporting intervals. When the consumer knows the schedule of the data reports, it can only detect missing data when the time expired.
The idea of the PCMC is that both participants can mutually detect if the communication between them is reliable, even if there are currently no critical data to be reported and processed.

PeterKranich committed:
Incorporated the comments from witkowb1 (Brian) which have been discussed at the WG meeting on 23-Oct-25.

marcagain committed:
update actors and transactions

Update actor description and transaction name/description
according to DCC agreement.
